packages/medusa/src/api/admin/orders/[id]/fulfillments/[fulfillment_id]/cancel/route.ts

```diff
@@ -6,16 +7,29 @@ import {
   AuthenticatedMedusaRequest,
   MedusaResponse,
 } from "../../../../../../../types/routing"
+import { AdminOrderCancelFulfillmentType } from "../../../../validators"
 
 export const POST = async (
-  req: AuthenticatedMedusaRequest,
+  req: AuthenticatedMedusaRequest<AdminOrderCancelFulfillmentType>,
   res: MedusaResponse
 ) => {
   const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
 
   const variables = { id: req.params.id }
 
-  // TODO: Workflow to cancel fulfillment + adjust inventory
+  const input = {
+    ...req.validatedBody,
+    order_id: req.params.id,
+  }
+
+  const { errors } = await cancelOrderFulfillmentWorkflow(req.scope).run({
```

Comment:
thought(non-blocking): I guess we could have a single Fulfillment Rest API and then depending on the input, run different workflows.

What I mean is, in the case of canceling a fulfillment, we could have:
```
POST /admin/fulfillments/:id/cancel
```

And then have mutually exclusive properties for each of the entities, that can be associated with a fulfillment, i.e. `order_id`, `exchange_id`, etc.

It would eliminate a few endpoints at the cost of having to do some work in the route to call the correct workflow. I am not sure what is best, but thought I'd share the idea.

Reply:
I see. The create would also behave the same? which is the current problem you catch. admin is not calling orders/:id/fulfillment to create it

Reply:
Yeah, I think we should be able to have one fulfillment Rest API for all operations. But we would still need to switch on the resource ID in the right and call the correct workflow, e.g. `createOrderFulfillmentWorkflow`, etc.

Reply:
Not necessarily something we have to do now. We can do it in a later PR.